### PR TITLE
cmd/utils: Add check on hard limit, skip test if below target

### PR DESCRIPTION
**Pull request by nolash**
Issue https://github.com/ethereum/go-ethereum/issues/15683

### CODE REVIEW

**Comment by karalabe** [comment]
This will fail building on Windows since file descriptor limits are handled differently there. I think it's fine to reduce the test target from 4K to 2K. That would solve your issue without requiring fancier changes.

Alternatively you could add a `getFdMaxLimit` function to the windows, unix and freebsd files to permt retrieving `syscall.Rlimit.Max`.

**Comment by nolash** [comment]
Thanks @karalabe , adjusted.

File: cmd/utils/fdlimit_test.go [hunk 1]
```diff
@@ -16,12 +16,24 @@
...
+	"fmt"
+	"syscall"
+	"testing"
+)
 
 // TestFileDescriptorLimits simply tests whether the file descriptor allowance
 // per this process can be retrieved.
 func TestFileDescriptorLimits(t *testing.T) {
 	target := 4096
+	var limit syscall.Rlimit
```
Comment (fjl) [inline cmd/utils/fdlimit_test.go]: Please avoid using the `syscall.Rlimit` type here to make the Windows build pass.

Comment (nolash) [reply cmd/utils/fdlimit_test.go]: argh sorry

**Pull request merged**